### PR TITLE
Use d2l-image component

### PR DESCRIPTION
**Pull request by JoshuaKlassen**
Update the user tile to consume the `d2l-image` element

See: https://github.com/Brightspace/d2l-image/pull/1

### CODE REVIEW

File: d2l-user-tile-auto.html [hunk 1]
```diff
@@ -100,7 +100,8 @@
 					if (image.class.indexOf('default-image') > -1) {
 						this._iconUrl = null;
 					} else {
-						this._iconUrl = (image.getLinkByRel(this.HypermediaRels.thumbnailRegular) || {}).href;
+						this._iconUrl = (image.getLinkByRel(this.HypermediaRels.thumbnailAlternative) || {}).href;
+						this.$.userTile.getProfileImage( this._iconUrl, this.token );
```
Comment (jstefaniuk-d2l) [inline d2l-user-tile-auto.html]: Rather doing this here, I'd pass the `iconUrl` and `token` parameters into `d2l-image` and have it set itself automatically based on what gets passed in

Comment (JoshuaKlassen) [inline d2l-user-tile-auto.html]: Are we concerned that somebody can just inspect element and see the token?

Comment (jstefaniuk-d2l) [inline d2l-user-tile-auto.html]: Nah, if they really wanted it they could just pull it off network requests, debug the javascript, or pull it off of the user-tile

File: bower.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
     "d2l-icons": "^3.0.0",
     "d2l-organization-hm-behavior": "git://github.com/Brightspace/organization-hm-behavior.git#^1.0.0",
     "iron-icon": "^1.0.0",
-    "d2l-tile": "git://github.com/Brightspace/d2l-tile#^1.0.2"
+    "d2l-tile": "git://github.com/Brightspace/d2l-tile#^1.0.2",
+    "d2l-image": "git://github.com/Brightspace/d2l-tile#^1.0.0"
```
Comment (ryantmer) [inline bower.json]: Should be `d2l-image` here ja?

File: d2l-user-tile.html [hunk 3]
```diff
@@ -75,7 +76,12 @@
 		<d2l-tile img-url="[[background]]" style$="[[_createBackgroundStyle(backgroundColor)]]">
 			<div class="user-tile-avatar">
 				<d2l-icon icon="navigation-48:profile" hidden$="{{!_hideCustomIcon(icon)}}" class="user-tile-default-icon"></d2l-icon>
-				<img src$="[[icon]]" hidden$="{{_hideCustomIcon(icon)}}" alt="" role="presentation" class="user-tile-avatar-image"/>
+				<div hidden$="{{_hideCustomIcon(icon)}}" class="user-tile-avatar-image">
+					<d2l-image
+						image-url$=[[icon]]
```
Comment (ryantmer) [inline d2l-user-tile.html]: Don't need the `$=` syntax here, that's only required for already-HTML-attributes (like `src` and `hidden`)

File: d2l-user-tile.html [hunk 4]
```diff
@@ -75,7 +76,12 @@
 		<d2l-tile img-url="[[background]]" style$="[[_createBackgroundStyle(backgroundColor)]]">
 			<div class="user-tile-avatar">
 				<d2l-icon icon="navigation-48:profile" hidden$="{{!_hideCustomIcon(icon)}}" class="user-tile-default-icon"></d2l-icon>
-				<img src$="[[icon]]" hidden$="{{_hideCustomIcon(icon)}}" alt="" role="presentation" class="user-tile-avatar-image"/>
+				<div hidden$="{{_hideCustomIcon(icon)}}" class="user-tile-avatar-image">
```
Comment (ryantmer) [inline d2l-user-tile.html]: The general rule of thumb I came across while investigating this a couple months ago is that if your hidden content has a web component within it, it's more performant to use `<template is="dom-if" if="[[blah]]">` than using the `hidden` attribute. I doubt there's much of a difference here, but worth being cognizant of.

**Comment by cgalvind2l** [comment]
Testing checklist can be seen here: https://github.com/Brightspace/d2l-image/pull/1

One issue with Edge but good besides that. This will be ready to merge once the PR in `d2l-image` is merged.

**Pull request merged**